pkg/controllers/provisioning/provisioner.go

```diff
@@ -100,32 +93,28 @@ func (p *Provisioner) Trigger() {
 	p.batcher.Trigger()
 }
 
-func (p *Provisioner) TriggerImmediate() {
-	p.batcher.TriggerImmediate()
-}
-
 func (p *Provisioner) Builder(_ context.Context, mgr manager.Manager) controller.Builder {
-	return controller.NewSingletonManagedBy(mgr).
-		WaitUntil(func(ctx context.Context) {
-			// Batch pods
-			p.batcher.Wait(ctx)
-
-			// wait to ensure that our cluster state is synced with the current known nodes to prevent over-provisioning
-			for WaitForClusterSync {
-				if err := p.cluster.Synchronized(ctx); err != nil {
-					logging.FromContext(ctx).Infof("waiting for cluster state to catch up, %s", err)
-					time.Sleep(1 * time.Second)
-				} else {
-					break
-				}
-			}
-		}).
-		WithOptions(controller.Options{
-			DisableWaitOnError: true,
-		})
+	return controller.NewSingletonManagedBy(mgr)
 }
 
-func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+func (p *Provisioner) Reconcile(ctx context.Context, _ reconcile.Request) (result reconcile.Result, err error) {
+	defer func() {
```

Comment:
Can we make a comment about why we re-trigger at the end of this provisioning loop?